nncf/torch/utils.py

```diff
@@ -299,24 +299,43 @@ def default_distributed_unwrapper(model: nn.Module):
         return model.module
     return model
 
+def rename_legacy_names_in_state_dict(state_dict_to_load: Dict[str, Any],
+                                      legacy_names: List[str],
+                                      legacy_name: str,
+                                      new_name: str):
+
+    for name in legacy_names:
+        tensor = state_dict_to_load.pop(name)
+        new_key = name.replace(legacy_name, new_name) if not new_name in name else name
+        state_dict_to_load[new_key] = tensor
+
+    if legacy_names:
+        warnings.warn('Legacy Batch Norm layer names was detected in checkpoint model state dict.'
+                      ' All occurrences of `{}` in nodes names was replaced by `{}`'.format(legacy_name, new_name),
+                      category=DeprecationWarning)
 
 def maybe_convert_legacy_names_in_model_state(state_dict_to_load: Dict[str, Any]) -> None:
     """
     Convert legacy layer names in compressed model state dict in case such names exist.
 
     :param state_dict_to_load: State dict to convert.
     """
-    legacy_bn_names = [name for name in state_dict_to_load if 'BatchNorm2d' in name]
-    for name in legacy_bn_names:
-        tensor = state_dict_to_load.pop(name)
-        new_name = name.replace('BatchNorm2d', 'NNCFBatchNorm')
-        state_dict_to_load[new_name] = tensor
-
-    if legacy_bn_names:
-        warnings.warn('Legacy Batch Norm layer names was detected in checkpoint model state dict.'
-                      ' All occurrences of `BatchNorm2d` in nodes names was replaced by `NNCFBatchNorm`',
-                      category=DeprecationWarning)
+    legacy_bn2d_names, legacy_bn1d_names, legacy_bn3d_names, legacy_nncfbn_names = [], [], [], []
```

Comment:
Don't save the vertical space in this way, it reduces readability.

Reply:
Removed.